src/contexts/DataContext.jsx

```diff
@@ -5,14 +5,15 @@ import { useEffect } from "react";
 const DataContext = createContext(null);
 
 export const DataProvider = ({ children }) => {
-
     const [notes, setNotes] = useState([]);
+    const [trashNotes, setTrashNotes] = useState([]);
+    const [archiveNotes, setArchiveNotes] = useState([]);
     const [currentNote, setCurrentNote] = useState({
```

Comment:
Suggestion: you can use reducer.

Reply:
i have move the state logic to reducer thanks for the suggestion